osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -79,6 +82,36 @@ private void load()
             Channel.PendingMessageResolved += pendingMessageResolved;
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            highlightedMessage = Channel.HighlightedMessage.GetBoundCopy();
+            highlightedMessage.BindValueChanged(_ => processMessageHighlighting(), true);
+        }
+
+        /// <summary>
+        /// Processes any pending message in <see cref="highlightedMessage"/>.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="CompositeDrawable.ScheduleAfterChildren"/> is for ensuring the scroll flow has updated with any new chat lines.
+        /// </remarks>
+        private void processMessageHighlighting() => ScheduleAfterChildren(() =>
+        {
+            if (highlightedMessage.Value == null)
+                return;
+
+            var chatLine = chatLines.SingleOrDefault(c => c.Message.Equals(highlightedMessage.Value));
+            if (chatLine == null)
+                return;
+
+            float center = scroll.GetChildPosInContent(chatLine, chatLine.DrawSize / 2) - scroll.DisplayableContent / 2;
+            scroll.ScrollTo(Math.Clamp(center, 0, scroll.ScrollableExtent));
+            chatLine.Highlight();
+
+            highlightedMessage.Value = null;
```

Comment:
I guess we're still nulling this in the end?

Reply:
Have mentioned one edge case of removing clearing this bindable in https://github.com/ppy/osu/pull/17137#discussion_r824005909.